### ISSUE TITLE
incompatible library version

### ISSUE THREAD

**Issue by Brentbin** [genome/civic-server#233]
Hi guys:
  I'm a new guy use ruby, and when I try to build civic-server there are some problem.
```
rake aborted!
LoadError: incompatible library version - /usr/local/rvm/gems/ruby-2.2.4/gems/pg-0.18.4/lib/pg_ext.so
/usr/local/rvm/gems/ruby-2.2.4/gems/pg-0.18.4/lib/pg.rb:4:in `require'
/usr/local/rvm/gems/ruby-2.2.4/gems/pg-0.18.4/lib/pg.rb:4:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.2.4/gems/bundler-1.13.1/lib/bundler/runtime.rb:91:in `require'
/usr/local/rvm/gems/ruby-2.2.4/gems/bundler-1.13.1/lib/bundler/runtime.rb:91:in `block (2 levels) in require'
/usr/local/rvm/gems/ruby-2.2.4/gems/bundler-1.13.1/lib/bundler/runtime.rb:86:in `each'
/usr/local/rvm/gems/ruby-2.2.4/gems/bundler-1.13.1/lib/bundler/runtime.rb:86:in `block in require'
/usr/local/rvm/gems/ruby-2.2.4/gems/bundler-1.13.1/lib/bundler/runtime.rb:75:in `each'
/usr/local/rvm/gems/ruby-2.2.4/gems/bundler-1.13.1/lib/bundler/runtime.rb:75:in `require'
/usr/local/rvm/gems/ruby-2.2.4/gems/bundler-1.13.1/lib/bundler.rb:106:in `require'
/git/civic-server/config/application.rb:7:in `<top (required)>'
/git/civic-server/Rakefile:4:in `require'
/git/civic-server/Rakefile:4:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/rake_module.rb:28:in `load'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/rake_module.rb:28:in `load_rakefile'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/application.rb:686:in `raw_load_rakefile'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/application.rb:96:in `block in load_rakefile'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/application.rb:178:in `standard_exception_handling'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/application.rb:95:in `load_rakefile'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/application.rb:79:in `block in run'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/application.rb:178:in `standard_exception_handling'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/lib/rake/application.rb:77:in `run'
/usr/local/rvm/gems/ruby-2.2.4/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.2.4/bin/rake:22:in `load'
/usr/local/rvm/gems/ruby-2.2.4/bin/rake:22:in `<main>'
/usr/local/rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `eval'
/usr/local/rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `<main>'
```
and my ruby version is 
```
ruby -v
ruby 2.2.4p230 (2015-12-16 revision 53155) [x86_64-linux]
```
howerver
```
 ldd /usr/local/rvm/gems/ruby-2.2.4/gems/pg-0.18.4/lib/pg_ext.so
	linux-vdso.so.1 =>  (0x00007ffd3e7ed000)
	libruby.so.2.0 => /usr/lib64/libruby.so.2.0 (0x00007fed8f777000)
	libpq.so.5 => /usr/lib64/libpq.so.5 (0x00007fed8f547000)
	libpthread.so.0 => /usr/lib64/libpthread.so.0 (0x00007fed8f32b000)
	libdl.so.2 => /usr/lib64/libdl.so.2 (0x00007fed8f127000)
	libcrypt.so.1 => /usr/lib64/libcrypt.so.1 (0x00007fed8eeef000)
	libm.so.6 => /usr/lib64/libm.so.6 (0x00007fed8ebed000)
	libc.so.6 => /usr/lib64/libc.so.6 (0x00007fed8e82b000)
	librt.so.1 => /usr/lib64/librt.so.1 (0x00007fed8e622000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fed8fdfd000)
	libssl.so.10 => /usr/lib64/libssl.so.10 (0x00007fed8e3b5000)
	libcrypto.so.10 => /usr/lib64/libcrypto.so.10 (0x00007fed8dfcd000)
	libkrb5.so.3 => /usr/lib64/libkrb5.so.3 (0x00007fed8dce7000)
	libcom_err.so.2 => /usr/lib64/libcom_err.so.2 (0x00007fed8dae3000)
	libgssapi_krb5.so.2 => /usr/lib64/libgssapi_krb5.so.2 (0x00007fed8d897000)
	libldap_r-2.4.so.2 => /usr/lib64/libldap_r-2.4.so.2 (0x00007fed8d63a000)
	libfreebl3.so => /usr/lib64/libfreebl3.so (0x00007fed8d437000)
	libk5crypto.so.3 => /usr/lib64/libk5crypto.so.3 (0x00007fed8d204000)
	libz.so.1 => /usr/lib64/libz.so.1 (0x00007fed8cfee000)
	libkrb5support.so.0 => /usr/lib64/libkrb5support.so.0 (0x00007fed8cddf000)
	libkeyutils.so.1 => /usr/lib64/libkeyutils.so.1 (0x00007fed8cbda000)
	libresolv.so.2 => /usr/lib64/libresolv.so.2 (0x00007fed8c9c0000)
	liblber-2.4.so.2 => /usr/lib64/liblber-2.4.so.2 (0x00007fed8c7b1000)
	libsasl2.so.3 => /usr/lib64/libsasl2.so.3 (0x00007fed8c593000)
	libssl3.so => /usr/lib64/libssl3.so (0x00007fed8c350000)
	libsmime3.so => /usr/lib64/libsmime3.so (0x00007fed8c129000)
	libnss3.so => /usr/lib64/libnss3.so (0x00007fed8be02000)
	libnssutil3.so => /usr/lib64/libnssutil3.so (0x00007fed8bbd6000)
	libplds4.so => /usr/lib64/libplds4.so (0x00007fed8b9d2000)
	libplc4.so => /usr/lib64/libplc4.so (0x00007fed8b7cc000)
	libnspr4.so => /usr/lib64/libnspr4.so (0x00007fed8b58e000)
	libselinux.so.1 => /usr/lib64/libselinux.so.1 (0x00007fed8b369000)
	libpcre.so.1 => /usr/lib64/libpcre.so.1 (0x00007fed8b107000)
	liblzma.so.5 => /usr/lib64/liblzma.so.5 (0x00007fed8aee2000)
```
when i change my ruby version from 2.2.4 to 2.0.0 to bundle install, error said ruby need to be ~>=2.1.0

Could anybody help me ?

**Comment by acoffman** [comment]
Hi @Brentbin,

Thanks for your interest in civic! The `pg` gem which provides the interface from ruby to postgres builds against several system libraries. It looks like the installation is failing due to some incompatibilities there. What OS are you using and what version of it? What version of postgres do you have installed? 

It looks like this issue here: https://bitbucket.org/ged/ruby-pg/issues/233/ruby-223-incompatible-library-version addresses what you're seeing.

**Comment by Brentbin** [comment]
THX for your answer!
My OS is Centos 7.2 and the version of my postgres is 
```
===========================================================================
 Package          Arch      Version   Repository                      Size
===========================================================================
Installing:
 pgdg-redhat94    noarch    9.4-1     /pgdg-redhat94-9.4-1.noarch    2.1 k

Transaction Summary
===========================================================================


===========================================================================
 Package                 Arch      Version                 Repository
                                                                      Size
===========================================================================
Installing:
 postgresql94-contrib    x86_64    9.4.9-1PGDG.rhel7       pgdg94    615 k
 postgresql94-server     x86_64    9.4.9-1PGDG.rhel7       pgdg94    3.9 M
Installing for dependencies:
 postgresql94            x86_64    9.4.9-1PGDG.rhel7       pgdg94    1.0 M
 postgresql94-libs       x86_64    9.4.9-1PGDG.rhel7       pgdg94    210 k

Transaction Summary
===========================================================================
Install  2 Packages (+2 Dependent packages)

Total download size: 5.7 M
Installed size: 25 M
Downloading packages:
(1/4): postgresql94-9.4.9-1PGDG.rhel7.x86_64.rpm      | 1.0 MB   00:02
(2/4): postgresql94-libs-9.4.9-1PGDG.rhel7.x86_64.rpm | 210 kB   00:00
(3/4): postgresql94-server-9.4.9-1PGDG.rhel7.x86_64.r | 3.9 MB   00:06
(4/4): postgresql94-contrib-9.4.9-1PGDG.rhel7.x86_64. | 615 kB   00:13
```

**Comment by acoffman** [comment]
We haven't tested CIViC on CentOS yet, only Debian based distros, but I don't see anything there that would indicate it wouldn't work. According to the comment found [here](https://bitbucket.org/ged/ruby-pg/issues/233/ruby-223-incompatible-library-version#comment-26731591) on that issue I linked earlier, it seems like the version of the postgres libraries that the gem was built with on installation differ from the versions of the libraries its trying to load at runtime. This could potentially happen if you'd changed your postgres version after running `bundle install` or if your library paths have changed. Beyond that, I'm not totally sure what might be causing it.

**Comment by malachig** [comment]
Hi @Brentbin, FYI, if you are just trying to take it for a test drive, here are some AWS instructions: https://github.com/genome/civic-server/wiki/Getting-Started-with-CIViC-Development-on-AWS

**Comment by Brentbin** [comment]
@acoffman @malachig 
THX for you great help!
I've fixed problem by upgrade Ruby from 2.2.3 to 2.3.0

**Comment by acoffman** [comment]
Awesome, glad you got it worked out!
